build.rs

```diff
@@ -136,14 +137,20 @@ fn main() -> std::io::Result<()> {
 
     println!("cargo:rustc-link-search={}", iceoryx_lib_dir);
 
+    println!("cargo:rustc-link-lib=iceoryx_posh_testing");
+
     println!("cargo:rustc-link-lib=iceoryx_posh_roudi");
     println!("cargo:rustc-link-lib=iceoryx_posh");
     println!("cargo:rustc-link-lib=iceoryx_hoofs");
     println!("cargo:rustc-link-lib=iceoryx_platform");
+
+    println!("cargo:rustc-link-lib=acl");
+
```

Comment:
@dkroenke for some weird reasons this is not required when building RouDi in iceoryx but it should be since libacl is linked as private in iceoryx_hoofs and ned specified as link labrary dor RouDi.

Since we export a header which includes an acl header, iceoryx_hoofs should export the libacl as public.